cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -488,6 +489,32 @@ class alignas(16) mutable_column_device_view
         detail::column_device_view_base::null_mask());
   }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Return first element (accounting for offset) after underlying data
+   * is casted to the specified type.
+   *
+   * @tparam T The desired type
+   * @return T* Pointer to the first element after casting
+   *---------------------------------------------------------------------------**/
+  template <typename T, std::enable_if_t<is_fixed_width<T>()> * = nullptr>
+  T* begin() const {
+    return data<T>();
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief This will always fail as begin can't be supported for non fixed width
+   * types.
+   *
+   * @throws `cudf::logic_error`, for non-fixed width types
+   *
+   * @tparam T The desired type
+   * @return T* Pointer to the first element after casting
+   *---------------------------------------------------------------------------**/
+  template <typename T, std::enable_if_t<not is_fixed_width<T>()> * = nullptr>
```

Comment:
Instead of doing this, just don't provide a path for the `not is_fixed_width` case, in which case if someone tries to do `mutable_column_device_view::begin<string_view>()` it will just fail to compile.